services/motion/builtin/move_request.go

```diff
@@ -895,6 +880,11 @@ func (ms *builtIn) createBaseMoveRequest(
 		obstaclePollingFreq = time.Duration(1000/motionCfg.obstaclePollingFreqHz) * time.Millisecond
 	}
 
+	// assign a planDeviationMM to our PlanRequest options so we know if we are already at the goal at plan time
+	valExtra.extra["planDeviationMM"] = motionCfg.planDeviationMM
```

Comment:
Why is this happening?

Reply:
since we check that we are not planning from the goal in the motionplan package we need some way of communicating the planDevMM value. Here we put it into the options field of the planrequest object